recipes/hdf5/all/conanfile.py

```diff
@@ -97,6 +94,11 @@ def validate(self):
            not self.options["szip"].enable_encoding:
             raise ConanInvalidConfiguration("encoding must be enabled in szip dependency (szip:enable_encoding=True)")
 
+    def validate_build(self):
+        if hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
```

Comment:
```suggestion
        if hasattr(self, "settings_build") and cross_building(self, skip_x64_x86=True):
```

Need to update cross_building, the old form `tools.cross_building` is deprecated. https://docs.conan.io/en/latest/reference/conanfile/tools/build.html#conan-tools-build-cross-building

Need to add the follow import:

`from conan.tools.build import cross_building`